src/Server.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace React\Http;
+
+use React\Http\Middleware\LimitConcurrentRequestsMiddleware;
+use React\Http\Middleware\RequestBodyBufferMiddleware;
+use React\Http\Middleware\RequestBodyParserMiddleware;
+use React\Socket\ServerInterface;
+
+/**
+ * Facade around StreamingServer with sane defaults for 80% of the use cases.
+ * The requests passed to your callable are fully compatible with PSR-7 because
+ * the body of the requests are fully buffered and parsed, unlike StreamingServer
+ * where the body is a raw ReactPHP stream.
+ *
+ * Wraps StreamingServer with the following middleware:
+ * - LimitConcurrentRequestsMiddleware
+ * - RequestBodyBufferMiddleware
+ * - RequestBodyParserMiddleware (only when enable_post_data_reading is true (default))
+ * - The callable you in passed as first constructor argument
+ *
+ * All middleware use their default configuration, which can be controlled with
+ * the the following configuration directives from php.ini:
+ * - upload_max_filesize
+ * - post_max_size
+ * - max_input_nesting_level
+ * - max_input_vars
+ * - file_uploads
+ * - max_file_uploads
+ * - enable_post_data_reading
+ */
+final class Server
+{
+    /**
+     * @var StreamingServer
+     */
+    private $streamingServer;
+
+    /**
+     * @see StreamingServer::__construct()
+     */
+    public function __construct($callback)
+    {
+        if (!is_callable($callback)) {
+            throw new \InvalidArgumentException();
+        }
+
+        $middleware = array();
+        $middleware[] = new LimitConcurrentRequestsMiddleware($this->getConcurrentRequestsLimit());
+        $middleware[] = new RequestBodyBufferMiddleware();
+        if (ini_get('enable_post_data_reading')) {
```

Comment:
This fails with HHVM and legacy PHP < 5.4: https://3v4l.org/qJtsa

```
$ php -d enable_post_data_reading=no -r 'var_dump(ini_get("enable_post_data_reading"));'
string(0) ""
```

Accordingly, this should probably check for `!== ''` instead? Also, can you add a comment for this?

Reply:
Hmmm interesting, but what happens then when set to `0`?

Reply:
According to http://php.net/manual/en/ini.core.php#ini.enable-post-data-reading it's a `boolean`.

Reply:
Updated the check